File: MatlabOctaveCodeSnippets/snip_transforms_non_orthogonal_basis.m
i=transpose([1,1]), j=transpose([0,1]) %non-orthogonal
x=3*i+2*j %create an arbitrary vector x to be analyzed
A=[i j]; %organize basis vectors as a matrix
temp=inv(A)*x; alpha=temp(1), beta=temp(2) %coefficients

File: PythonCodeSnippets/snip_transforms_non_orthogonal_basis.py
import numpy as np
from numpy.linalg import inv

# Non-orthogonal
i = np.transpose([[1, 1]])
j = np.transpose([[0, 1]])  

x = 3 * i + 2 * j  # Create an arbitrary vector x to be analyzed
A = np.concatenate((i, j), axis=1)# Organize basis vectors as a matrix

temp = np.dot(inv(A), x)
# Coefficients
alpha = temp[0]
beta = temp[1]
